### PR TITLE
Newsletters: add 249 (2023-05-03)

### PR DESCRIPTION
**Pull request by harding**
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin Core 25158 @bitschmidty 
- [x] Bitcoin Core 26933 @glozow 
- [x] Bitcoin Core 25325 @Xekyo

### CODE REVIEW

File: _posts/en/newsletters/2023-05-03-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    As of this writing, Ingala's post had not received any direct
+    replies, but there was [continued discussion][halseth matt] about
+    his original proposal for MATT and it's ability to allow
+    verification that an (essentially) arbitrarily complex program was
+    run.
+
+- **Analysis of signature adaptor security:** Adam Gibson
+  [posted][gibson adaptors] to the Bitcoin-Dev mailing list an analysis
+  of the security of [signature adaptors][topic adaptor signatures],
+  especially about how they interact with [multisignatures][topic
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-05-03-newsletter.md:43]: Maybe

```suggestion
  especially about how they interact with [multisignature][topic
```

File: _posts/en/newsletters/2023-05-03-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  avoid the risk of zero-fee transactions losing their fee-bumping
+  descendant in the event of a replacement. It may reversed in the
+  future if a DoS-resistant method of preventing such transactions is
+  found, e.g. through a package topology restriction like v3 or a
+  modification to the mempool's eviction process.
+
+- [Bitcoin Core #25325][] introduces a pool based memory resource for
+  the UTXO cache. The new data structure pre-allocates and manages a
+  larger pool of memory to track UTXOs instead of allocating and freeing
+  memory for each UTXO individually. UTXO lookups cause a major part of
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-05-03-newsletter.md:104]: Maybe

```suggestion
  memory for each UTXO individually. UTXO lookups represent a major proportion of
```

File: _posts/en/newsletters/2023-05-03-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  is stored in Bitcoin Core's UTXO set, the `utxoupdatepsbt` RPC can add
+  the necessary scriptPubKey and amount data to any PSBT spending a
+  UTXO.  The `utxoupdatepsbt` also previously searched the local node's
+  mempool for UTXOs to allow users to spend outputs from unconfirmed
+  transactions.
+
+  However, after `utxoupdatepsbt` was added to Bitcoin Core, several
+  hardware signing devices began requiring even spends of segwit v0
+  outputs to include full transactions in order to prevent a variation
+  on fee overpayment that could result from a user seemingly signing the
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-05-03-newsletter.md:135]:
```suggestion
  of fee overpayment that could result from a user seemingly signing the
```

**Comment by harding** [comment]
Made all edits for @bitschmidty feedback (thanks!), reviewed additions by @bitschmidty, @glozow, @Xekyo (thanks!), added lede, updated releases/RCs, and add topic page entries.  Thanks everyone!

File: _topics/en/psbt.md [hunk 4]
```diff
@@ -195,6 +195,9 @@ optech_mentions:
   - title: "LND #7122 adds support for importing PSBTs from binary files"
     url: /en/newsletters/2022/11/30/#lnd-7122
 
+  - title: "Bitcoin Core #25939 allows nodes with txindex enabled to add txes with `utxoupdatepsbt` RPC"
```
Comment (bitschmidty) [inline _topics/en/psbt.md:198]:
```suggestion
  - title: "Bitcoin Core #25939 allows nodes with txindex enabled to add txs with `utxoupdatepsbt` RPC"
```

**Comment by bitschmidty** [comment]
One small fixup in the topic review. Squashed and merged! Thanks @harding @Xekyo @glozow for contributing this week!

**Pull request merged**